src/main/java/bio/terra/service/common/gcs/GcsUriUtils.java

```diff
@@ -0,0 +1,88 @@
+package bio.terra.service.common.gcs;
+
+import bio.terra.service.dataset.exception.InvalidUriException;
+import org.apache.commons.lang3.StringUtils;
+
+public final class GcsUriUtils {
+  public static class GsUrlParts {
+    private String bucket;
```

Comment:
I'm not sure blobid is what we want, it uses the name of the blob whereas we use path of the blob

Reply:
I'm pretty sure Blob's `name` is the same thing as our `path`. It may be odd to think of the GS path as a name, but most bucket type stores like GS and S3 are really key/value stores, and the path or name is the key.